Homework/Nam_Nguyen/Lesson11/LeetCode_14_4Sum.java

```diff
@@ -0,0 +1,45 @@
+package Homework.Nam_Nguyen.Lesson11;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class LeetCode_14_4Sum {
+
+    public List<List<Integer>> fourSum(int[] nums, int target) {
+        List<List<Integer>> res = new ArrayList<>();
+        int len = nums.length;
+        Arrays.sort(nums);
+
+        for (int i = 0; i < len - 1; i++) {
+            for (int j = i + 1; j < len; j++) {
+                int k = target - nums[i] - nums[j];
+                int left = j + 1;
+                int right = len - 1;
+                while(left < right) {
+                    if(nums[left] + nums[right] == k) {
+                        if(!res.contains(Arrays.asList(nums[i],nums[j],nums[left],nums[right]))){
```

Comment:
what is runtime of contains function in Java?

Reply:
@quangh33 It is O(n^4) where n equals 200 (the length of nums array). I tried to use Set instead of List to avoid using contain() function and that's a bit better but I clearly need some extra space.